src/App.tsx

```diff
@@ -16,39 +16,35 @@ import Brightness4Icon from '@mui/icons-material/Brightness4';
 import Brightness7Icon from '@mui/icons-material/Brightness7';
 import useMediaQuery from '@mui/material/useMediaQuery';
 
-import { switchTheme, saveState, LIGHT, DARK } from './redux/appSlice';
+import { switchTheme, Theme, saveTheme, fetchTheme } from './redux/appSlice';
+import { fetchTodos } from './redux/todosSlice';
 import { useDispatch, useSelector } from 'react-redux';
-import { RootState } from './redux/store';
+import { RootState, AppDispatch } from './redux/store';
 
 const ColorModeContext = createContext({
   toggleColorMode: (mode: 'light' | 'dark' | undefined) => {},
 });
 
 const App = () => {
-  const dispatch = useDispatch();
+  const dispatch: AppDispatch = useDispatch();
 
   const mode = useSelector((state: RootState) => state.config.mode);
 
   const prefersColorScheme = useMediaQuery('(prefers-color-scheme: dark)')
-    ? DARK
-    : LIGHT;
+    ? Theme.DARK
+    : Theme.LIGHT;
 
   useEffect(() => {
```

Comment:
Decompose your app. Don't be afraid of creating multiple React hooks to encapsulate your logic

Reply:
Do you mean to create two useEffect hooks with one dispatch call inside instead of one useEffect hook with two dispatch calls?

Reply:
What you are looking for is a hook like `useTheme`, which will encapsulate everything there is to theming

Reply:
useTheme and useTodos hooks have been created